Assets/GothicVR/Scripts/Creator/WaynetCreator.cs

```diff
@@ -29,8 +29,10 @@ private void CreateWaypoints(GameObject parent, WorldData world)
 
             foreach (var waypoint in world.waypoints)
             {
-                var wpobject = GameObject.CreatePrimitive(PrimitiveType.Cube);
+                // var wpobject = GameObject.CreatePrimitive(PrimitiveType.Cube);
+                var wpobject = new GameObject();
```

Comment:
Why do we need this? Because of reasons for spawning?

If we need it for this only, I would recommend we hold xyz coordinates in an array only. Having 5000 additional GameObjects just for getting Vector3 position might be a performance issue for Unity. (As we already have hundreds of GOs)

Reply:
we also need the rotation for teleporting the player